### PR TITLE
save curvilinear covariance as a vector (size 15)

### PR DESCRIPTION
**Pull request by slava77**
this is used to make trackingNtuple for https://github.com/trackreco/mkFit/pull/325

### CODE REVIEW

**Comment by slava77** [comment]
I was planning to make a PR for CMSSW with what's accumulated in trackreco:CMSSW_11_2_0_mkFit_X and this PR.

File: Validation/RecoTrack/plugins/TrackingNtuple.cc [hunk 1]
```diff
@@ -3260,29 +3200,13 @@ void TrackingNtuple::fillSeeds(const edm::Event& iEvent,
...
-        see_stateCcov44.push_back(stateCcov[4][4]);
-        see_stateCcov45.push_back(stateCcov[4][5]);
-        see_stateCcov55.push_back(stateCcov[5][5]);
+      if (addSeedCurvCov_) {
+        auto const& stateCcov = tsos.curvilinearError().matrix();
+        std::vector<float> cov(15);
+        auto covP = cov.begin();
+        for (auto const val : stateCcov)
+          *(covP++) = val;//row-major
+        see_stateCurvCov.push_back(cov);
```
Comment (makortel) [inline Validation/RecoTrack/plugins/TrackingNtuple.cc:3209]: In principle
```suggestion
        see_stateCurvCov.emplace_back(std::move(cov));
```
would avoid a copy.

**Comment by makortel** [comment]
Looks ok, I'm not sure how important avoiding a copy in this code is. I could merge either way.

File: Validation/RecoTrack/plugins/TrackingNtuple.cc (hunk 1 above)
Comment (slava77) [reply Validation/RecoTrack/plugins/TrackingNtuple.cc]: doesn't this get optimized to the same result (especially considering the few-line scope)?

**Comment by slava77** [comment]
> Looks ok, I'm not sure how important avoiding a copy in this code is. I could merge either way.

OK, I updated and also squashed/rebased

File: Validation/RecoTrack/plugins/TrackingNtuple.cc (hunk 1 above)
Comment (makortel) [reply Validation/RecoTrack/plugins/TrackingNtuple.cc]: Godbolt shows difference (fewer assembly lines with `std::move`)), and without `std::move()` standard requires to make a copy (IIUC). `push_back` vs. `emplace_back` showed no difference.

**Pull request merged**